lib/utils.h

```diff
@@ -98,6 +98,23 @@ inline static const char *getOrdinalText(int i)
     return "th";
 }
 
+inline std::string encodeXml(const std::string& data) 
```

Comment:
I think we should remove `ErrorLogger::toxml` and use this instead. but this function needs some additional handling we have in `ErrorLogger::toxml` also.